octoprint_mrbeam/analytics/analytics_handler.py

```diff
@@ -40,80 +82,339 @@ def _subscribe(self):
+					if 'high_precision' in payload:
+						del payload['high_precision']
+					if 'markers_recognized' in payload:
+						del payload['markers_recognized']
+
+				if payload is not None:
+					data[ak.DATA] = payload
+
+				self._write_event(ak.CAM_EVENT, eventname, self._cam_event_log_version, payload=data)
+		except Exception as e:
+			self._logger.error('Error during write_cam_event: {}'.format(e.message))
+
+	def _write_event(self, typename, eventname, version, payload=None):
+		try:
+			data = {
+				ak.SERIALNUMBER: self._getSerialNumber(),
+				ak.TYPE: typename,
+				ak.VERSION: version,
+				ak.EVENT: eventname,
+				ak.TIMESTAMP: time.time()
+			}
+			if payload is not None:
+				data.update(payload)
+			self._append_data_to_file(data)
+		except Exception as e:
+			self._logger.error('Error during _write_event: {}'.format(e.message))
+
+	def add_dust_value(self, dust_value):
+		"""
+		:param dust_value:
+		:return:
+		"""
+		if self._analyticsOn and self._current_dust_collector is not None:
+			try:
+				self._current_dust_collector.addValue(dust_value)
+			except Exception as e:
+				self._logger.error('Error during add_dust_value: {}'.format(e.message))
+
+	def add_laser_temp_value(self,laser_temp):
+		"""
+		:param laser_temp:
+		:return:
+		"""
+		if self._analyticsOn and self._current_lasertemp_collector is not None:
+			try:
+				self._current_lasertemp_collector.addValue(laser_temp)
+			except Exception as e:
+				self._logger.error('Error during add_laser_temp_value: {}'.format(e.message))
+
+	def add_laser_intensity_value(self, laser_intensity):
+		"""
+		Laser intensity.
+		:param laser_intensity: 0-255. Zero means laser is off
+		"""
+		if self._analyticsOn and self._current_intensity_collector is not None:
+			try:
+				self._current_intensity_collector.addValue(laser_intensity)
+			except Exception as e:
+				self._logger.error('Error during add_laser_intensity_value: {}'.format(e.message))
+
+	def write_dust_log(self, data):
+		try:
+			if self._analyticsOn:
+				self._write_jobevent(ak.FINAL_DUST,payload=data)
+		except Exception as e:
+			self._logger.error('Error during write dust_log: {}'.format(e.message))
+
+	def _init_jsonfile(self):
+		open(self._jsonfile, 'w+').close()
 		data = {
-			'type':'dust',
-			'v':1,
-			'dust_start':values['dust_start'],
-			'dust_end': values['dust_end'],
-			'dust_start_ts': values['dust_start_ts'],
-			'dust_end_ts': values['dust_end_ts']
+			ak.HOSTNAME: self._getHostName(),
+			ak.SERIALNUMBER: self._getSerialNumber(),
+			ak.LASERHEAD_VERSION: self._getLaserHeadVersion()
 		}
-		self._append_data_to_file(data)
+		self._write_deviceinfo(ak.INIT,payload=data)
+		self._write_current_software_status()
 
 	def _append_data_to_file(self, data):
-		with open(self._jsonfile, 'a') as f:
-			json.dump(data, f)
-			f.write('\n')
+		try:
+			if not os.path.isfile(self._jsonfile):
+				self._init_jsonfile()
+			with open(self._jsonfile, 'a') as f:
+				json.dump(data, f)
```

Comment:
Could we pls get the json as string, append the newline and then write it in one pseudo-atomic operation. Appears to be more fail safe than this.
(I know, not your code, Clemens)